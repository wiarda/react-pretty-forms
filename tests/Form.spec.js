/* global describe test expect */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  Form,
  FormField,
  FormButton,
  FormCheckbox,
  FormSelect,
  IfResolved,
  IfFailed,
  IfSubmitting,
  IfActive,
} from '../index';

describe('Form component', () => {
  test('Matches snapshot', () => {
    const component = renderer.create(<Form action="blank" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders required props', () => {
    const component = renderer.create(<Form action="blank" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Accepts custom styles', () => {
    const component = renderer
      .create(<Form action="blank" cssModule={{ abc: { fontSize: '1rem' } }} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders a full form', () => {
    const component = renderer
      .create(
        <Form action="blank">
          <div>plain type</div>
          <IfActive>
            <FormSelect
              required
              name="salutation"
              label="Preferred salutation"
              defaultValue="unselected"
            >
              <option value="">-</option>
              <option value="Dr.">Dr.</option>
              <option value="Prof.">Prof.</option>
            </FormSelect>
            <FormField
              type="text"
              name="first"
              label="First name*"
              required
              initialValue="Gortok"
            />
            <FormField type="text" name="last" label="Last name*" required initialValue="Hamfist" />
            <FormField type="text" name="role" label="Job Title" />
            <FormField type="text" name="institution" label="Institution*" required />
            <FormField type="email" name="email" label="Email*" required />
            <FormCheckbox
              label="Here is a rather long label that might cause some clipping issues"
              required
            />
            <FormField type="hidden" name="source" initialValue="direct" />
            <FormButton type="submit" value="Book a Meeting" />
          </IfActive>
          <IfResolved>Resolved...</IfResolved>
          <IfSubmitting>Loading...</IfSubmitting>
          <IfFailed>Failed...</IfFailed>
        </Form>,
        {
          createNodeMock: element => {
            if (element.type === 'select') {
              return {
                current: {
                  value: 'test value',
                },
              };
            }
          },
        },
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
