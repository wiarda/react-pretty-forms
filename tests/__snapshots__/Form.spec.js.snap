// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Accepts custom styles 1`] = `
<form
  action="blank"
  className="pretty-form"
  data-formstatus="active"
  encType=""
  method="post"
  onChange={[Function]}
/>
`;

exports[`Form component Matches snapshot 1`] = `
<form
  className="pretty-form"
  data-formstatus="active"
  encType=""
  method="post"
  onChange={[Function]}
/>
`;

exports[`Form component Renders a full form 1`] = `
<form
  action="blank"
  className="pretty-form"
  data-formstatus="active"
  encType=""
  method="post"
  onChange={[Function]}
>
  <div
    styles={Object {}}
  >
    plain type
  </div>
  <div
    className="pretty-active"
    data-formstatus="active"
  >
    <div
      className="pretty-row"
    >
      <label
        className="pretty-label"
        data-active={true}
        data-validity={true}
        htmlFor="salutation"
      >
        <span
          className="pretty-label-text"
          data-active={true}
        >
          Preferred salutation
        </span>
        <select
          className="form--pretty-select"
          defaultValue="unselected"
          multiple={false}
          name="salutation"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          required={true}
          size={1}
        >
          <option
            disabled={true}
            hidden={true}
            value="unselected"
          />
          <option
            value=""
          >
            -
          </option>
          <option
            value="Dr."
          >
            Dr.
          </option>
          <option
            value="Prof."
          >
            Prof.
          </option>
        </select>
      </label>
      <div
        className="pretty-validation"
        data-validity={true}
      >
        Please fill in this required field
      </div>
    </div>
    <div
      className="pretty-row"
      style={Object {}}
    >
      <label
        className="pretty-label"
        data-active={true}
        data-validity={true}
        htmlFor="first"
      >
        <span
          className="pretty-label-text"
          data-active={true}
        >
          First name*
        </span>
        <input
          className="pretty-input"
          data-validity={true}
          defaultValue="Gortok"
          name="first"
          onBlur={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
      </label>
      <div
        className="pretty-validation"
        data-validity={true}
      >
        Please fill in this required field
      </div>
    </div>
    <div
      className="pretty-row"
      style={Object {}}
    >
      <label
        className="pretty-label"
        data-active={true}
        data-validity={true}
        htmlFor="last"
      >
        <span
          className="pretty-label-text"
          data-active={true}
        >
          Last name*
        </span>
        <input
          className="pretty-input"
          data-validity={true}
          defaultValue="Hamfist"
          name="last"
          onBlur={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
      </label>
      <div
        className="pretty-validation"
        data-validity={true}
      >
        Please fill in this required field
      </div>
    </div>
    <div
      className="pretty-row"
      style={Object {}}
    >
      <label
        className="pretty-label"
        data-active={false}
        data-validity={true}
        htmlFor="role"
      >
        <span
          className="pretty-label-text"
          data-active={false}
        >
          Job Title
        </span>
        <input
          className="pretty-input"
          data-validity={true}
          name="role"
          onBlur={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </label>
      <div
        className="pretty-validation"
        data-validity={true}
      >
        
      </div>
    </div>
    <div
      className="pretty-row"
      style={Object {}}
    >
      <label
        className="pretty-label"
        data-active={false}
        data-validity={true}
        htmlFor="institution"
      >
        <span
          className="pretty-label-text"
          data-active={false}
        >
          Institution*
        </span>
        <input
          className="pretty-input"
          data-validity={true}
          name="institution"
          onBlur={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
      </label>
      <div
        className="pretty-validation"
        data-validity={true}
      >
        Please fill in this required field
      </div>
    </div>
    <div
      className="pretty-row"
      style={Object {}}
    >
      <label
        className="pretty-label"
        data-active={false}
        data-validity={true}
        htmlFor="email"
      >
        <span
          className="pretty-label-text"
          data-active={false}
        >
          Email*
        </span>
        <input
          className="pretty-input"
          data-validity={true}
          name="email"
          onBlur={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
        />
      </label>
      <div
        className="pretty-validation"
        data-validity={true}
      >
        Please enter a valid email address
      </div>
    </div>
    <div
      className="pretty-row"
    >
      <label
        className="pretty-checkbox-label"
        data-validity={true}
        htmlFor="checkbox"
      >
        <span
          className="pretty-checkbox-label-text"
        >
          Here is a rather long label that might cause some clipping issues
        </span>
        <input
          className="pretty-checkbox"
          data-validity={true}
          id="checkbox"
          name="checkbox"
          onClick={[Function]}
          type="checkbox"
          value="Yes"
        />
      </label>
      <div
        className="pretty-validation"
        data-validity={true}
      >
        Please click to accept
      </div>
    </div>
    <div
      className="pretty-row"
      style={
        Object {
          "height": "0",
          "margin": "unset",
          "padding": "unset",
          "visibility": "hidden",
        }
      }
    >
      <input
        className="pretty-input"
        data-validity={true}
        defaultValue="direct"
        name="source"
        onBlur={[Function]}
        onFocus={[Function]}
        required={false}
        type="hidden"
      />
      ;
      <div
        className="pretty-validation"
        data-validity={true}
      >
        
      </div>
    </div>
    <input
      className="pretty-button"
      onClick={[Function]}
      style={Object {}}
      type="submit"
      value="Book a Meeting"
    />
  </div>
</form>
`;

exports[`Form component Renders required props 1`] = `
<form
  action="blank"
  className="pretty-form"
  data-formstatus="active"
  encType=""
  method="post"
  onChange={[Function]}
/>
`;
